### PR TITLE
Add data explained pdf to help menu

### PR DESCRIPTION
**Pull request by aleekaup2**

### CODE REVIEW

**Comment by jussiks** [comment]
Add the pdf to `potku.spec` so it will be included in the standalone executable:

```python
a = Analysis(
    ["potku.py"],
     pathex=[],
     binaries=bins,
     datas=[
        ("documentation/Potku_data_explained.pdf", "documentation/"),
        ("external/bin/jibal.conf", "external/bin/"),
         ... etc. ...
```

File: potku.py [hunk 1]
```diff
@@ -1485,6 +1486,28 @@ def __open_manual(self):
                 self, "Not found",
                 "There is no manual to be found!",
                 QtWidgets.QMessageBox.Ok, QtWidgets.QMessageBox.Ok)
+                
+                
+    def __open_data_help(self):
+        """Open data help file.
+        """
+        # TODO duplicate code with __open_manual
```
Comment (jussiks) [inline potku.py:1494]: Yup. Refactor the duplicated code into a single function that takes a file path to be opened as an argument.

File: potku.py [hunk 2]
```diff
@@ -1485,6 +1486,28 @@ def __open_manual(self):
...
+        # TODO changed the file path to point to the manual, I guess this needs
+        #      to be updated in the .spec file too?
+        manual_filename = Path("documentation", "Potku_data_explained.pdf")
```
Comment (jussiks) [inline potku.py:1497]: This path may not work in Mac application. You could change it to
```python
file_name = gf.get_root_dir() / "documentation" / "Potku_data_explained.pdf"
```
(add `import modules.general_functions as gf` if the module is not already imported)

File: potku.py [hunk 3]
```diff
@@ -1469,22 +1471,37 @@ def __tab_exists(self, tab_id):
     def __open_manual(self):
         """Open user manual.
         """
+        manual_filename = gf.get_root_dir() / "documentation" / "Potku-User-Manual.pdf"
+        self.__open_file(manual_filename)        
+                
+                
+    def __open_data_help(self):
```
Comment (jussiks) [inline potku.py:1478]:
```suggestion
        self.__open_file(manual_filename)        

    def __open_data_help(self):
```
Code style fix:
- PEP 8: E303 too many blank lines (2)

**Comment by jussiks** [comment]
@mlaitin or @jaakkojulin should take a look at this PR. Should the PDF be included with Potku?

**Comment by jaakkojulin** [comment]
Binary files and other files that can not be easily modified or files that can be generated from other data should not be included in the repository.

There is a need for a binary bundle of files that are needed on top of version controlled code to build the distributable (and usable) versions. Of course the Git repo could contain some reference to them (name, SHA hash etc). I don't know if there are tools to facilitate this.

For my CMake projects (e.g. JIBAL), I have thought of using CMake's [ExternalData](https://cmake.org/cmake/help/latest/module/ExternalData.html). Maybe something similar is has been made for Python too. This way all large files that can not be "compiled" by the developers easily could be stored on e.g. on a web server.

Documentation should not rely on individual PDFs either.

**Review by samivout — APPROVED** [review]
The changes work as intended and the new data manual button is functional both in the development version and build version.

**Comment by samivout** [comment]
@mlaitin As noted in the review comment, the changes are functional, but for the time being I don't have write permissions to Potku's repository, so I'm unable to merge this pull request to master.

**Pull request merged**